src/web/services/course.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { MessageOutput } from '../types/api-output';
+import { CourseArchiveRequest, CourseCreateRequest, CourseSaveRequest } from '../types/api-request';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles sessions related logic provision.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class CourseService {
+
+  constructor(private httpRequestService: HttpRequestService) {
+  }
+
+  /**
+   * Creates a course by calling API.
+   */
+  createCourse(paramMap: { [key: string]: string }, request: CourseCreateRequest): Observable<MessageOutput> {
+    return this.httpRequestService.post('/course', paramMap, request);
+  }
+
+  /**
+   * Updates a course by calling API.
+   */
+  updateCourse(paramMap: { [key: string]: string }, request: CourseSaveRequest): Observable<MessageOutput> {
```

Comment:
You should have something like `updateCourse(courseId: string, request: CourseSaveRequest)`?

Related:

> * Note that the create and update calls don't actually have use for the parameter querystring, but it's still there to be passed in an empty list.

Just remove the parameter.